diff -urNp setup-2.8.14-orig/profile setup-2.8.14/profile
--- setup-2.8.14-orig/profile	2010-12-09 13:58:46.617505004 +0100
+++ setup-2.8.14/profile	2010-12-09 13:59:50.623239254 +0100
@@ -56,7 +56,7 @@ export PATH USER LOGNAME MAIL HOSTNAME H
 
 for i in /etc/profile.d/*.sh ; do
     if [ -r "$i" ]; then
-        if [ "$PS1" ]; then
+        if [ "${-#*i}" != "$-" ]; then
             . "$i"
         else
             . "$i" >/dev/null 2>&1
