diff -urNp setup-2.8.14-orig/bashrc setup-2.8.14/bashrc
--- setup-2.8.14-orig/bashrc	2010-12-09 13:43:55.303492223 +0100
+++ setup-2.8.14/bashrc	2010-12-03 16:25:58.000000000 +0100
@@ -26,14 +26,14 @@ if [ "$PS1" ]; then
         if [ -e /etc/sysconfig/bash-prompt-xterm ]; then
             PROMPT_COMMAND=/etc/sysconfig/bash-prompt-xterm
         else
-            PROMPT_COMMAND='echo -ne "\033]0;${USER}@${HOSTNAME%%.*}:${PWD/#$HOME/~}"; echo -ne "\007"'
+            PROMPT_COMMAND='printf "\033]0;%s@%s:%s\007" "${USER}" "${HOSTNAME%%.*}" "${PWD/#$HOME/~}"'
         fi
         ;;
     screen)
         if [ -e /etc/sysconfig/bash-prompt-screen ]; then
             PROMPT_COMMAND=/etc/sysconfig/bash-prompt-screen
         else
-            PROMPT_COMMAND='echo -ne "\033_${USER}@${HOSTNAME%%.*}:${PWD/#$HOME/~}"; echo -ne "\033\\"'
+            PROMPT_COMMAND='printf "\033]0;%s@%s:%s\033\\" "${USER}" "${HOSTNAME%%.*}" "${PWD/#$HOME/~}"'
         fi
         ;;
     *)
