diff -urNp setup-2.8.14-orig/bashrc setup-2.8.14/bashrc
--- setup-2.8.14-orig/bashrc	2016-11-01 15:25:49.096138018 +0100
+++ setup-2.8.14/bashrc	2016-11-01 15:46:04.695682919 +0100
@@ -62,7 +62,7 @@ if ! shopt -q login_shell ; then # We're
     # Current threshold for system reserved uid/gids is 200
     # You could check uidgid reservation validity in
     # /usr/share/doc/setup-*/uidgid file
-    if [ $UID -gt 199 ] && [ "`id -gn`" = "`id -un`" ]; then
+    if [ $UID -gt 199 ] && [ "`/usr/bin/id -gn`" = "`/usr/bin/id -un`" ]; then
        umask 002
     else
        umask 022
diff -urNp setup-2.8.14-orig/csh.cshrc setup-2.8.14/csh.cshrc
--- setup-2.8.14-orig/csh.cshrc	2016-11-01 15:25:49.088137969 +0100
+++ setup-2.8.14/csh.cshrc	2016-11-01 15:48:10.533092536 +0100
@@ -7,7 +7,7 @@
 # Current threshold for system reserved uid/gids is 200
 # You could check uidgid reservation validity in
 # /usr/share/doc/setup-*/uidgid file
-if ($uid > 199 && "`id -gn`" == "`id -un`") then
+if ($uid > 199 && "`/usr/bin/id -gn`" == "`/usr/bin/id -un`") then
     umask 002
 else
     umask 022
@@ -20,7 +20,7 @@ if ($?prompt) then
     # make completion work better by default
     set autolist
   else
-    set prompt=\[$user@`hostname -s`\]\$\ 
+    set prompt=\[$user@`/bin/hostname -s`\]\$\ 
   endif
 endif
 
diff -urNp setup-2.8.14-orig/profile setup-2.8.14/profile
--- setup-2.8.14-orig/profile	2016-11-01 15:25:49.108138092 +0100
+++ setup-2.8.14/profile	2016-11-01 15:47:40.646995252 +0100
@@ -25,10 +25,10 @@ pathmunge () {
 if [ -x /usr/bin/id ]; then
     if [ -z "$EUID" ]; then
         # ksh workaround
-        EUID=`id -u`
-        UID=`id -ru`
+        EUID=`/usr/bin/id -u`
+        UID=`/usr/bin/id -ru`
     fi
-    USER="`id -un`"
+    USER="`/usr/bin/id -un`"
     LOGNAME=$USER
     MAIL="/var/spool/mail/$USER"
 fi
@@ -58,7 +58,7 @@ export PATH USER LOGNAME MAIL HOSTNAME H
 # Current threshold for system reserved uid/gids is 200
 # You could check uidgid reservation validity in
 # /usr/share/doc/setup-*/uidgid file
-if [ $UID -gt 199 ] && [ "`id -gn`" = "`id -un`" ]; then
+if [ $UID -gt 199 ] && [ "`/usr/bin/id -gn`" = "`/usr/bin/id -un`" ]; then
     umask 002
 else
     umask 022
